apm-agent-plugins/apm-kafka-plugin/apm-kafka-spring-plugin/src/main/java/co/elastic/apm/agent/kafka/SpringKafkaBatchListenerInstrumentation.java

```diff
@@ -0,0 +1,100 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.kafka;
+
+import co.elastic.apm.agent.impl.transaction.Transaction;
+import co.elastic.apm.agent.sdk.logging.Logger;
+import co.elastic.apm.agent.sdk.logging.LoggerFactory;
+import co.elastic.apm.agent.util.LoggerUtils;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+import javax.annotation.Nullable;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+
+/**
+ * Spring internally does the batch processing in two steps:
+ * 1. Iterate over all Messages and store them in a List
+ * 2. Pass the list to the registered batch KafkaListener
+ * <p>
+ * In order to catch both operations in a single transaction with span links,
+ * this instrumentation creates a transaction which spans both of the actions listed above.
+ */
+public class SpringKafkaBatchListenerInstrumentation extends BaseKafkaInstrumentation {
+
+    @Override
+    public ElementMatcher<? super TypeDescription> getTypeMatcher() {
+        return named("org.springframework.kafka.listener.KafkaMessageListenerContainer$ListenerConsumer");
+    }
+
+    @Override
+    public ElementMatcher<? super MethodDescription> getMethodMatcher() {
+        return named("invokeBatchListener")
+            .and(takesArguments(1))
+            .and(takesArgument(0, named("org.apache.kafka.clients.consumer.ConsumerRecords")));
+    }
+
+    @Override
+    public String getAdviceClassName() {
+        return getClass().getName() + "$SpringKafkaBatchListenerAdvice";
+    }
+
+    public static class SpringKafkaBatchListenerAdvice {
+
+        private static final Logger oneTimeTransactionCreationWarningLogger;
+
+        static {
+            oneTimeTransactionCreationWarningLogger = LoggerUtils.logOnce(LoggerFactory.getLogger("Spring-Kafka-Batch-Logger"));
+        }
+
+        @Nullable
+        @Advice.OnMethodEnter(suppress = Throwable.class, inline = false)
+        public static Object onEnter(@Advice.This Object thiz) {
+            Transaction transaction = tracer.startRootTransaction(thiz.getClass().getClassLoader());
+            if (transaction != null) {
+                transaction
+                    .withType("messaging")
+                    .withName("Spring Kafka Message Batch Processing")
```

Comment:
[minor] I think here using a simpler naming like `Spring Kafka Batch` would be enough to keep things short. The `Message` is quite implicit from `Kafka`, and the `Processing` does not add any relevant information.

Reply:
As discussed on slack: We will stay with the current name to keep it consistent with the spring AMQP batch listener transactions.